### DISCUSSION TITLE
v2.0.0 Technical Preview 8

### DISCUSSION THREAD

**Discussion by the1812** [the1812/Bilibili-Evolved#2282] [分享 / Share]
**功能:**

- 组件与 `v1.12.18` 同步更新.
- 使用在线仓库时, 连接到 `github.com` 时无需再设置跨域.
- 修复搜索栏建议的文本过长时溢出.
- 快捷键设置也能从搜索栏中唤起了.
- 修复播放器控制栏的扩展按钮(截图, 逐帧调整等)颜色不对.
- 在组件详情的菜单中, 鼠标停留在 `检查更新` 上时可以显示安装的来源 URL, 如果是来自 `localhost`, 还会显示特定的图标作为区分. (#2278)
- 快捷键支持 <kbd>Ctrl</kbd>+<kbd>Enter</kbd> 发送评论. (#1015)

**开发:**

- 新增 `RuntimeLibrary API` (`src/core/runtime-library.ts`), 可以在运行时动态载入第三方库, 可以避免在每次载入页面时都加载了不常用的库. 以下原先内置于脚本的库现已转换为运行时库:
  - protobufjs
  - JSZip
  - Sortable
- 重新整理了 `task.json` 中任务的命名.
- 功能也和本体一样能参与 CI 构建流程了.
- 支持记录第三方功能并在在线仓库中显示.
- 在 `CONTRIBUTING.md` 中补充了一些 API 说明.
- 主题颜色除了 10 级不同透明度 `var(--theme-color-XX)`, 也支持了 10 级不同明亮度 `var(--theme-color-lightness-XX)`.
- `Toast` 的 `duration` 即使在 Toast 已发出后也能响应更改了.
- 同 v1 一样, 按住 <kbd>Shift</kbd> 点击侧栏的功能将会执行 `debugger` 语句, 在开发者工具打开时能够立即停止页面运行.
- 清理了一些无用文件, `widgets` 文件夹中的代码移动到 `ui` 和 `components` 中对应的文件夹.
- `observer` 新增 `urlChange`, 可以检测 URL 变动.

<hr /><em>This discussion was created from the release <a href='https://github.com/the1812/Bilibili-Evolved/releases/tag/v2.0.0-tp8'>v2.0.0 Technical Preview 8</a>.</em>

**Comment by the1812** [comment]
下一次 TP9 可能是最后一轮 TP 了, 之后就会进入预览版了

**Comment by kethoven** [comment]
希望点击自定义顶栏上的“动态”“稍后再看”“收藏”能够在新的标签页打开，而不是覆盖当前页面。

**Reply by Geeyun-JY3** [reply]
要不让点击自定义顶栏和原版顶栏各栏目的打开方式都能自定义？

**Reply by the1812** [reply]
我是想给自定义顶栏一个总体设置去选, 习惯新标签页打开的话应该希望顶栏上所有链接都是新标签页打开的吧, 不用逐个设置

**Reply by Geeyun-JY3** [reply]
就是不知道有没有用户有原版顶栏的需求：Logo 当前标签页打开，剩余其他新标签页打开

**Reply by the1812** [reply]
那就加个隐藏设置, 按照顶栏项 name 覆盖新标签页打开设置, 这样也方便默认设置对齐原版
```js
options.openInNewTab = true
options.openInNewTabOverrides = { logo: false }
```

**Comment by kethoven** [comment]
目前发现bug：自定义顶栏的动态，会停留在上次刷新出来的状态。出了新的动态也刷不出来。
